src/extension.ts

```diff
@@ -534,9 +534,26 @@ export class ExtensionManager implements vscode.Disposable {
 
     private async initActiveProject(): Promise<CMakeProject | undefined> {
         let folder: vscode.WorkspaceFolder | undefined;
-        if (vscode.workspace.workspaceFolders && vscode.window.activeTextEditor && this.workspaceConfig.autoSelectActiveFolder) {
-            folder = vscode.workspace.getWorkspaceFolder(vscode.window.activeTextEditor.document.uri);
-            await this.updateActiveProject(folder, vscode.window.activeTextEditor);
```

Comment:
It feels like this change is overly complex for what we need to do.  I would think you just need to check if `folder` has a value. If not, then set `folder` to `vscode.workspace.workspaceFolderes[0]` and `editor` to `undefined` like you did below.

It looks like you are reimplementing `getWorkspaceFolder` with that for loop.